src/test/java/com/gooddata/http/client/GoodDataHttpClientIntegrationTest.java

```diff
@@ -5,40 +5,61 @@
+import static net.jadler.Jadler.initJadler;
+import static net.jadler.Jadler.onRequest;
+import static net.jadler.Jadler.port;
+import static net.jadler.Jadler.verifyThatRequest;
+
+import net.jadler.Request;
+import net.jadler.stubbing.RequestStubbing;
+import net.jadler.stubbing.Responder;
+import net.jadler.stubbing.ResponseStubbing;
+import net.jadler.stubbing.StubResponse;
+
 import org.apache.http.HttpHost;
-import org.apache.http.HttpResponse;
 import org.apache.http.HttpStatus;
 import org.apache.http.client.HttpClient;
-import org.apache.http.client.methods.HttpGet;
-import org.apache.http.entity.ContentType;
-import org.apache.http.impl.client.HttpClientBuilder;
-import org.apache.http.util.EntityUtils;
 import org.junit.After;
 import org.junit.Before;
-import org.junit.Ignore;
 import org.junit.Test;
 
 import java.io.IOException;
 import java.nio.charset.Charset;
-
-import static net.jadler.Jadler.closeJadler;
-import static net.jadler.Jadler.initJadler;
-import static net.jadler.Jadler.onRequest;
-import static net.jadler.Jadler.port;
-import static org.junit.Assert.assertEquals;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
 
 public class GoodDataHttpClientIntegrationTest {
 
-    private static final String GDC_TOKEN_URL = "/gdc/account/token";
-    public static final String GDC_LOGIN_URL = "/gdc/account/login";
-    public static final String GDC_PROJECTS_URL = "/gdc/projects";
-    public static final String REDIRECT_URL = "/redirect/to/projects";
+    private static final String GDC_TOKEN_PATH = "/gdc/account/token";
+    private static final String GDC_LOGIN_PATH = "/gdc/account/login";
+    private static final String GDC_PROJECTS_PATH = "/gdc/projects";
+    private static final String GDC_PROJECTS2_PATH = "/gdc/projects2";
+    private static final String REDIRECT_PATH = "/redirect/to/projects";
+
     private static final String SST_HEADER = "X-GDC-AuthSST";
+    private static final String WWW_AUTHENTICATE_HEADER = "WWW-Authenticate";
+    private static final String ACCEPT_HEADER = "Accept";
+    private static final String CONTENT_HEADER = "Content-Type";
+    private static final String TT_HEADER = "X-GDC-AuthTT";
+
+    private static final String GOODDATA_REALM = "GoodData realm=\"GoodData API\"";
+    private static final String TT_COOKIE = "cookie=GDCAuthTT";
+
+    private static final String BODY_401 = "<html><head><title>401 Authorization Required</title></head><body><p>This server could not verify that you are authorized to access the document requested.  Either you supplied the wrong credentials (e.g., bad password), or your browser doesn't understand how to supply the credentials required.Please see <a href=\"http://docs.gooddata.apiary.io/#login\">Authenticating to the GoodData API</a> for details.</p></body></html>";
```

Comment:
It's true the tests work without bodies, but isn't it usefull only for reader to show, how the communication looks in real?